From a1cb0ea501c5c968352fee385339713dd25a8a9f Mon Sep 17 00:00:00 2001
From: Dan Walsh <dwalsh@redhat.com>
Date: Thu, 8 May 2014 10:14:00 -0400
Subject: [PATCH] Device mapper was creating an invalid cgroup name.

This was breaking docker top CONTAINER

Docker-DCO-1.1-Signed-off-by: Daniel Walsh <dwalsh@redhat.com> (github: rhatdan)
---
 daemon/execdriver/native/driver.go | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/daemon/execdriver/native/driver.go b/daemon/execdriver/native/driver.go
index 59d193e..f183e29 100644
--- a/daemon/execdriver/native/driver.go
+++ b/daemon/execdriver/native/driver.go
@@ -193,6 +193,9 @@ func (d *driver) GetPidsForContainer(id string) ([]int, error) {
 	filename := filepath.Join(cgroupRoot, cgroupDir, id, "tasks")
 	if _, err := os.Stat(filename); os.IsNotExist(err) {
 		filename = filepath.Join(cgroupRoot, cgroupDir, "docker", id, "tasks")
+		if _, err := os.Stat(filename); os.IsNotExist(err) {
+			filename = filepath.Join(cgroupRoot, cgroupDir, fmt.Sprintf("docker-%s.scope", id), "tasks")
+		}
 	}
 
 	output, err := ioutil.ReadFile(filename)
-- 
1.9.3

